pkg/dataobj/logs_reader.go

```diff
@@ -109,33 +113,77 @@ func (r *LogsReader) Read(ctx context.Context, s []Record) (int, error) {
 		}
 	}
 
-	r.buf = slices.Grow(r.buf, len(s))
+	r.buf = slicegrow.GrowToCap(r.buf, len(s))
 	r.buf = r.buf[:len(s)]
 
+	// Fill the row buffer with empty values so they can re-use the memory we pass in.
+	for i := range r.buf {
```

Comment:
nit: this could be extracted so it's only called when necessary. This is because `*LogsReader.Read` is repeatedly called with the same buffer. We can take advantage of this to minimize re-initialization of `r.buf`:
```
bufLen := len(r.buf)
sLen := len(s)
if bufLen < sLen {
  slicegrow.GrowToCap
  r.populate(r.buf[bufLen:])
}
```

pseudo-code, but hopefully it gets the idea across.